.github/workflows/main.yml

```diff
@@ -2,77 +2,55 @@ name: Run Code Checks
 
 on:
   push:
+    branches-ignore: [develop]
     paths:
       - '**'
       - '!docker/**'
       - '!.github/workflows/statistician-*'
       - '!notebooks/**'
       - '!docs/**'
+      - '!k8s/**'
       - '!README.md'
+  workflow_run:
+    workflows: ["Create a new tag"]
+    types:
+      - completed
 
 jobs:
   build-wheels:
+    if: |
+      github.event_name == 'push' || (github.ref == 'refs/heads/develop'
```

Comment:
Would `github.ref == 'refs/heads/develop'` ever be true considering you have `branches-ignore: [develop]`?

Reply:
`on` -- events operates in `or` logic, so the workflow here will be trigger either by `push` or `workflow_run`. If `push` happens on `develop`, it won't be triggered. Otherwise if `workflow_run` happens, it means a pr got merged and the commit got tagged, then the workflow should be triggered on `develop` since there is no restriction on branches for `workflow_run` such that it can build a "release" version based on the tag on `develop` branch.

Reply:
which implies, only pr merge will trigger a release but the commits caused by pr merge won't, that was I intended for.

Reply:
Makes sense! In that case it's all good by me